modules/internal/ChapelBase.chpl

```diff
@@ -632,9 +629,24 @@ module ChapelBase {
     }
   }
 
+  inline proc postfix!(type t:unmanaged) type {
+    return _to_nonnil(t);
+  }
+  inline proc postfix!(type t:borrowed) type {
+    return _to_nonnil(t);
+  }
+
+  inline proc postfix!(x:unmanaged!) {
+    return _to_nonnil(x);
+  }
+  inline proc postfix!(x:borrowed!) {
+    checkNotNil(x);
```

Comment:
The above line is a very strange spot for `checkNotNil`. Also, it's not done for `unmanaged!` even though the two cases should be similar I think. So please double-check.